course7/homeworks/patrickkostjens/chat/chat_client.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <websocketpp/config/asio_no_tls_client.hpp>
+#include <websocketpp/client.hpp>
+#include <websocketpp/common/thread.hpp>
+#include "njson/json.hpp"
+
+#include <string>
+
+typedef websocketpp::client<websocketpp::config::asio_client> client;
+using json = nlohmann::json;
+
+class ChatClient {
+public:
+    ChatClient (const std::string& uri);
+    ~ChatClient();
+
+    void run();
+
+private:
+    void on_open(websocketpp::connection_hdl hdl);
+    void on_message(websocketpp::connection_hdl, client::message_ptr msg);
+    void send_json(const json& obj);
+    void handle_handshake_reply(const json& obj);
+
+    client m_endpoint;
+    websocketpp::lib::thread m_thread;
+    websocketpp::connection_hdl m_hdl;
+
+    bool m_open;
+    bool m_is_registered;
```

Comment:
These are not initialized and therefore did not work on my machine when testing.

Reply:
Good point, not sure why they didn't cause me problems, but I'll fix it.